views-jstachio/src/main/java/io/micronaut/views/jstachio/JStachioMessageBodyWriter.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.jstachio;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+
+import io.jstach.jstachio.JStachio;
+import io.jstach.jstachio.Output;
+import io.jstach.jstachio.Template;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.type.Argument;
+import io.micronaut.core.type.MutableHeaders;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.annotation.Produces;
+import io.micronaut.http.body.MessageBodyWriter;
+import io.micronaut.http.codec.CodecException;
+import jakarta.inject.Singleton;
+
+/**
+ * A message body writer for JStachio JStache annotated models.
+ * If the {@link JStachio} supports the body type it writes the template.
+ * If the HTTP Header Content-Type is not present, it adds it to the response HTTP Headers with the value text/html.
+ * @param <T> type to be encoded
+ * @author agentgt
+ * @since 4.1.0
+ */
+@Singleton
+@Produces({MediaType.TEXT_HTML})
+public class JStachioMessageBodyWriter<T> implements MessageBodyWriter<T> {
+    private static final String HTML_UTF_8 = MediaType.TEXT_HTML + "; charset=UTF-8";
+    private final JStachio jstachio;
+
+    /**
+     *
+     * @param jstachio JStachio
+     */
+    public JStachioMessageBodyWriter(JStachio jstachio) {
+        this.jstachio = jstachio;
+    }
+
+    @Override
+    public boolean isWriteable(@NonNull Argument<T> type, @Nullable MediaType mediaType) {
```

Comment:
isWriteable will not always be considered. But it seems to be required here?

Reply:
Sorry, do you mean that the contents of `writeTo` should check whether `isWriteable` returns `true`?

```
    @Override
    public void writeTo(@NonNull Argument<T> type,
                        @NonNull MediaType mediaType,
                        T object,
                        @NonNull MutableHeaders outgoingHeaders,
                        @NonNull OutputStream outputStream) throws CodecException {
if ( isWriteable(type, mediatype) {
dologic
}
```

Reply:
I added a [test which verifies it plays well with ViewsFilter](https://github.com/micronaut-projects/micronaut-views/pull/590/commits/0fdb62d3686b42f486f2c5b04beb22306acc6e03) and a [test which verifies how it behaves with other HTML MessageBodyWriter](https://github.com/micronaut-projects/micronaut-views/pull/590/commits/8ee799b1fe24f8e3d582e4aa6cc2de1d48ae3379) and a [related core's pr ](https://github.com/micronaut-projects/micronaut-core/pull/9789).

Reply:
thanks